File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/models/History/SensorTypeData.java
package egovframework.base.history.ibll.models.history;

public class SensorTypeData {

    private int sensorTypeCode = -1;
    private Integer sensorSubTypeNo = null;
    private String sensorTypeName = "";

    public int getSensorTypeCode() {
        return sensorTypeCode;
    }

    public void setSensorTypeCode(int sensorTypeCode) {
        this.sensorTypeCode = sensorTypeCode;
    }

    public Integer getSensorSubTypeNo() {
        return sensorSubTypeNo;
    }

    public void setSensorSubTypeNo(Integer sensorSubTypeNo) {
        this.sensorSubTypeNo = sensorSubTypeNo;
    }

    public String getSensorTypeName() {
        return sensorTypeName;
    }

    public void setSensorTypeName(String sensorTypeName) {
        this.sensorTypeName = sensorTypeName;
    }

    public long getKey() {
        return makeKey(sensorTypeCode, sensorSubTypeNo);
    }

    public static long makeKey(int sensorTypeNo, Integer sensorSubTypeNo) {
        if (sensorSubTypeNo == null) {
            return (long) sensorTypeNo;
        }

        long hi = ((long) sensorTypeNo) << 32;
        long low = sensorSubTypeNo.longValue();
        return hi | low;
    }
}

File: src/SOP8/Base/SDMS/Base.SDMS.IBLL/Models/SensorTypeData.cs
namespace Base.SDMS.IBLL.Models
{
    public class SensorTypeData
    {
        private int m_nSensorTypeCode = -1;
        private int? m_sensorSubTypeNo = null;
        private string m_strSensorTypeName = "";

        public int SensorTypeCode
        {
            get { return m_nSensorTypeCode; }
            set { m_nSensorTypeCode = value; }
        }

        public int? SensorSubTypeNo
        {
            get { return m_sensorSubTypeNo; }
            set { m_sensorSubTypeNo = value; }
        }

        public string SensorTypeName
        {
            get { return m_strSensorTypeName; }
            set { m_strSensorTypeName = value; }
        }

        public long GetKey()
        {
            return MakeKey(m_nSensorTypeCode, m_sensorSubTypeNo);
        }

        public static long MakeKey(int sensorTypeNo, int? sensorSubTypeNo)
        {
            if (sensorSubTypeNo == null)
            {
                return (long)sensorTypeNo;
            }

            long hi = ((long)sensorTypeNo << 32);
            long low = (long)sensorSubTypeNo;
            return (hi | low);
        }
    }
}
